src/tasks/barfTasks.ts

```diff
@@ -343,15 +344,55 @@ const turns: AdventureAction[] = [
       setChoice(536, choiceToSet);
       ensureEffect($effect`Transpondent`);
       use($item`Map to Safety Shelter Grimace Prime`);
-      return true;
     },
+    completed: () => true,
     spendsTurn: true,
     sobriety: Sobriety.DRUNK,
   },
+  {
+    name: "Thesis",
+    ready: needTurns,
+    do: deliverThesisIfAble,
+    completed: () => !needTurns(),
+    sobriety: Sobriety.SOBER,
+    spendsTurn: false,
```

Comment:
The logic here is too involved for me to type out here, but this is not strictly true; I think it is true iff we have the NEP open and haven't used all 10 adventures. So I guess it's not that involved.
```suggestion
    spendsTurn: () => !(canAdventure($location`The Neverending Party`) && get("_neverendingPartyFreeTurns") < 10),
```